src/jit/compiler.hpp

```diff
@@ -2768,7 +2769,28 @@ inline bool Compiler::fgIsThrowHlpBlk(BasicBlock* block)
         return false;
     }
 
-    GenTreePtr call = block->bbTreeList->gtStmt.gtStmtExpr;
+    GenTree* call = nullptr;
+    if (block->IsLIR())
+    {
+        LIR::Range& blockRange = LIR::AsRange(block);
+        call                   = blockRange.LastNode();
```

Comment:
We do have that: see `BasicBlock::lastNode()`. I can refactor this code a bit in order to use it.